quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -105,6 +106,10 @@ ConfigValue getOrDefault(String name, ConfigSourceInterceptorContext context, Co
             // if not defined, return the current value from the property as a default if the property is not explicitly set
             if (defaultValue == null
                     || (current != null && !current.getConfigSourceName().equalsIgnoreCase("default values"))) {
+                if (defaultValue == null && mapper != null) {
```

Comment:
this.mapper = mapper == null ? PropertyMapper::defaultTransformer : mapper;
=> mapper is always defined, so we could remove the `&& mapper != null` part of the condition here, couldn't we?

Reply:
not really, it can be null.

Reply:
i'd argue you would "want" to set the transformer to null then, but okay either way ;)

Reply:
oh, just seen that the defaultTransformer only returns the value, which could be null... forget what I just said, then it's fine :)